server/src/main/java/com/broadcom/lsp/cobol/service/delegates/completions/SectionCompletion.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package com.broadcom.lsp.cobol.service.delegates.completions;
+
+import com.broadcom.lsp.cobol.service.CobolDocumentModel;
+import com.broadcom.lsp.cobol.service.delegates.validations.AnalysisResult;
+import com.google.inject.Singleton;
+import lombok.NonNull;
+import org.eclipse.lsp4j.CompletionItemKind;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+
+import static com.broadcom.lsp.cobol.service.delegates.completions.CompletionOrder.SECTIONS;
+
+/**
+ * Provide completion functionality for sections
+ */
+@Singleton
+public class SectionCompletion implements Completion {
+
+  @NonNull
+  @Override
+  public Collection<String> getCompletionSource(CobolDocumentModel document) {
```

Comment:
@NonNull is non-consistent here, because of  ofNullable (yes empty value returned).  Moreover as in SectionLocations class you can add: @NonNull CobolDocumentModel

Reply:
So should I remove all `@NonNull` statements for `getCompletionSource` for all `Completion` classes?

Reply:
We do not return the Optional, we just use it to wrap the given value. CobolDocumentModel here is nullable

Reply:
> So should I remove all `@NonNull` statements for `getCompletionSource` for all `Completion` classes?

For this method yes, but for  other methods they look fine

Reply:
> > So should I remove all `@NonNull` statements for `getCompletionSource` for all `Completion` classes?
> 
> For this method yes, but for other methods they look fine

Why? Let them stay there

Reply:
> em stay t

It has no affect here, for this situation

Reply:
OK! Removed :-)